contracts/src/LightClient.sol

```diff
@@ -362,40 +361,44 @@ contract LightClient is Initializable, OwnableUpgradeable, UUPSUpgradeable {
         uint256 updatesCount = stateHistoryCommitments.length;
 
         // Handling Edge Cases
-        // Edgecase 1: The block is in the future or
-        // before HotShot was live, allow for at least two updates before considering HotShot live
+        // Edgecase 1: blockNumber is greater than current block.number or
+        // less than 3 updates exist which means the history is insufficient to determine a lag.
         if (blockNumber > block.number || updatesCount < 3) {
             revert InsufficientSnapshotHistory();
         }
 
         uint256 prevBlock;
         bool prevUpdateFound;
 
+        // Start searching from the latest update
         uint256 i = updatesCount - 1;
         while (!prevUpdateFound) {
-            if (stateHistoryCommitments[i].l1BlockHeight <= blockNumber) {
-                prevUpdateFound = true;
-                prevBlock = stateHistoryCommitments[i].l1BlockHeight;
-            }
-
-            // We don't consider the lag time for the first two updates
+            // Skip the first two updates as per logic mentioned above.
             if (i < 2) {
                 break;
             }
 
-            // We've reached the first recorded block
-            if (i == stateHistoryFirstIndex) {
+            // Stop if we've already assessed the first recorded state in the history.
+            if (i < stateHistoryFirstIndex) {
```

Comment:
should this be  `if (i <= stateHistoryFirstIndex)`

Reply:
not "=" because I still want to evaluate when the index  is equal to the stateHistoryFirstIndex